src/providers/trusted_service/context/mod.rs

```diff
@@ -48,6 +49,59 @@ mod key_management;
 )]
 mod ts_protobuf {
     include!(concat!(env!("OUT_DIR"), "/ts_crypto.rs"));
+
+    pub trait OpcodeT {
```

Comment:
I've implemented this and the `SetHandle` below so that the whole "open key -> do operation -> close key" isn't done in each individual method (and also so that you don't have to pass the opcode along with the request - each request can only have one specific opcode anyway).